lib/msf/core/module_manager/loading.rb

```diff
@@ -22,6 +22,9 @@ module Msf::ModuleManager::Loading
   ]
 
   def file_changed?(path)
+    if  path =~ %r{scanner/msmail/host_id} || path =~ %r{gather/office365userenum}
```

Comment:
TODO: Pairing with Chris showed that the `gather/office365userenum` file gets loaded on start up, as it's missing from the module metadata cache.

This explains why the user's log files are often filled with these messages:
```
[01/21/2021 14:43:33] [e(0)] core: Unable to load module /Users/adfoster/Documents/code/metasploit-framework/modules/auxiliary/gather/office365userenum.py - LoadError  Try running file manually to check for errors or dependency issues.
```

I believe this problem is happening on Jenkins for the same reason users can't load this module on most 'fresh' setups, not having the required dependencies:

> ➜  metasploit-framework git:(improves-handling-off-external-modules-missing-runtime-dependencies) ✗ PYTHONPATH=./lib/msf/core/modules/external/python:$PYTHONPATH python ./modules/auxiliary/gather/office365userenum.py                                 [12/01/21 | 3:14:31]
Missing Dependency! python-requests required!
> {"params": {"message": "\n\n.       .1111...          | Title: office365userenum.py\n    .10000000000011.   .. | Author: Oliver Morton (Sec-1 Ltd)\n .00              000...  | Email: oliverm@sec-1.com\n1                  01..   | Description:\n                    ..    | Enumerate valid usernames from Office 365 using\n                   ..     | ActiveSync.\nGrimHacker        ..      | Requires: Python 2.7 or 3.6, python-requests\n                 ..       |\ngrimhacker.com  ..        |\n@grimhacker    ..         |\n----------------------------------------------------------------------------\n    This program comes with ABSOLUTELY NO WARRANTY.\n    This is free software, and you are welcome to redistribute it\n    under certain conditions. See GPLv2 License.\n----------------------------------------------------------------------------\n", "level": "info"}, "jsonrpc": "2.0", "method": "message"}

This is a separate issue from this pull request, just calling it out so it's not forgotten about.

Reply:
Opened an separate issue for this: https://github.com/rapid7/metasploit-framework/issues/14705